File: plugins/java-decompiler/testData/Enum.java
public enum Enum {
  E1,
  E2() {
    @Override
    public void m() { }
  },
  E3("-"),
  E4("+") {
    @Override
    public void m() { }
  };

  public void m() { }

  private String s;

  private Enum() { this("?"); }
  private Enum(@Deprecated String s) { this.s = s; }
}

File: plugins/kotlin/idea/tests/testData/quickfix/deprecatedSymbolUsage/enum.kt
// "Replace with 'BAR'" "true"
// K2_AFTER_ERROR: UNRESOLVED_IMPORT
// K2_AFTER_ERROR: UNRESOLVED_REFERENCE

enum class Enm {
    @Deprecated("Replace with BAR", ReplaceWith("BAR"))
    FOO,
    BAR
}

fun test() {
    Enm.FOO<caret>
}
// IGNORE_K2